forem/forem
# pullToRefresh adjustments - fixing scrolling hijacking

## What type of PR is this? (check all applicable)


## Description

We've been noticing some issues with horizontally scrollable elements on touch devices, like the most recent one: editor toolbar. It was just a little "picky" - sometimes working and sometimes not working. I've built a little [prototype in codepen](https://codepen.io/ludwiczakpawel/full/eYGmxGj) to validate whether it's me going crazy or is this really something related to our codebase...

Anyway, what initially felt like some [weird CSS bug](https://github.com/forem/forem/pull/15615), turned out to be caused by our pullToRefresh library which is hijacking some touch events (that's why it's been happening only on touch devices). This library is primarily useful only on Forem iOS app but not really in actual browsers (Safari, Chrome), since all the browsers support "pull to refresh" behavior natively. 

As a short-term solution, we've decided to disable this library on `/new` view (only) since it's pointless there anyway. This fixed the issue with editor toolbar but it's still a problem elsewhere (for example, we have scrollable tabs on the Search Results view, on mobile). As an addition I've also slightly adjusted this library settings making this behavior a tiny bit better (but **not** eliminating the problem entirely). And that's what this PR does.

As for long-term solution, I think we should (cc @fdoxyz):

1. Disable pullToRefresh library on actual web view but keep it enabled for Forem iOS app if needed.
2. Drop this library entirely and rely on some native solution for Forem iOS app.

## Related Tickets & Documents

- [This slack thread](https://forem-team.slack.com/archives/C7GE84DEJ/p1638390238247900).
- [Related PR](https://github.com/forem/forem/pull/15615) that I thought gonna fix the issue.

## QA Instructions, Screenshots, Recordings

1. Here's how to review PRs involving actual mobile devices: https://forem.team/fdoxyz/how-to-review-prs-with-the-mobile-experience-in-mind-13mg (I highly recommend ngrok solution).
2. Once it's done, just navigate to /new and see if scrolling behavior on editor toolbar performs slightly better than it is right now on production `dev.to/new`. Also feel free to test whether pullToRefresh works as expected on other views (since it will be disabled on `/new`)

### UI accessibility concerns?

If anything, this improves the accessibility.

## Added/updated tests?

- [x] No need. 

## [Forem core team only] How will this change be communicated?

- [x] No need.
